### DISCUSSION TITLE
Install should shallow clone?

### DISCUSSION THREAD

**Discussion by tengelmeier** [Homebrew/discussions#2862] [Getting started] [answered]
Hi, 

I just set up a new machine and first-time installed brew. It took ages - is it really necessary to download 440MB package data if I want only install the latest versions of current repos? I'd assume a shallow clone with a depth of the tap repo would suffice?

==> Tapping homebrew/core
remote: Enumerating objects: 1132961, done.
remote: Counting objects: 100% (117/117), done.
remote: Compressing objects: 100% (64/64), done.
remote: Total 1132961 (delta 66), reused 99 (delta 53), pack-reused 1132844
Receiving objects: 100% (1132961/1132961), 440.55 MiB | 450.00 KiB/s, done.
Resolving deltas: 100% (782554/782554), done.

**Comment by SMillerDev** [comment]
Yes it is otherwise the required compute power would bring down GitHub. (We got complaints from github when we tried).

**Comment by SMillerDev** [comment]
If you're looking for a way to save the storage equivalent of a large website, you can always try the experimental `HOMEBREW_INSTALL_FROM_API` variable in https://docs.brew.sh/Manpage

**Comment by carlocab** [comment, marked as answer]
See also:
https://github.com/Homebrew/discussions/discussions/226
https://github.com/Homebrew/discussions/discussions/225
https://github.com/Homebrew/discussions/discussions/1738
https://github.com/Homebrew/discussions/discussions/231
https://github.com/Homebrew/brew/issues/9426
https://github.com/Homebrew/brew/pull/9383
https://github.com/Homebrew/brew/issues/10557
https://github.com/Homebrew/brew/issues/11693
https://github.com/Homebrew/brew/issues/7050

**Reply by SMillerDev** [reply]
Pinned the discussion with this answer for further reference
